product_weight_through_uom/models/product_template.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 Tecnativa S.L. - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    extra_weight = fields.Float(
+        'Extra Weight',
+        compute='_compute_extra_weight',
+        inverse='_inverse_extra_weight',
+        digits=dp.get_precision('Stock Weight'),
+        help="Extra weight (in Kg.) given by packaging, etc.",
+    )
+    is_weight_uom = fields.Boolean(
+        compute='_compute_is_weight_uom',
+        readonly=True,
+        default=False,
+    )
+
+    @api.depends('product_variant_ids', 'product_variant_ids.weight')
+    def _compute_extra_weight(self):
+        unique_variants = self.filtered(
+            lambda x: len(x.product_variant_ids) == 1)
+        for template in unique_variants:
+            template.extra_weight = template.product_variant_ids.extra_weight
+        for template in (self - unique_variants):
+            template.extra_weight = 0.0
+
+    @api.multi
+    def _inverse_extra_weight(self):
+        for template in self.filtered(
+                lambda x: len(x.product_variant_ids) == 1):
+            template.product_variant_ids.extra_weight = template.extra_weight
+
+    @api.depends('uom_id.category_id')
```

Comment:
```suggestion
    @api.depends('uom_id', 'uom_id.category_id')
```
It's more important to take into account that someone changes UoM in the product that a category is changed in the UoM :wink: